sql/moz-fx-data-shared-prod/telemetry_derived/urlbar_search_sessions_v1/query.sql

```diff
@@ -0,0 +1,178 @@
+WITH events_unnested AS (
+  SELECT
+    DATE(submission_timestamp) AS submission_date,
+    sample_id,
+    client_info.client_id,
+    timestamp AS event_timestamp,
+    normalized_channel,
+    normalized_country_code,
+    name AS event_name,
+    CASE
+      WHEN name = 'engagement'
+        AND (
+          (
+            mozfun.map.get_key(extra, "engagement_type") IN (
+              "click",
+              "drop_go",
+              "enter",
+              "go_button",
+              "paste_go"
+            )
+          )
+        )
+        THEN 'engaged'
+      WHEN name = 'abandonment'
+        THEN 'abandoned'
+      WHEN name = 'engagement'
+        AND (
+          (
+            mozfun.map.get_key(extra, "engagement_type") NOT IN (
+              "click",
+              "drop_go",
+              "enter",
+              "go_button",
+              "paste_go"
+            )
+          )
+        )
+        THEN "annoyance"
+      ELSE NULL
+    END AS search_session_type,
+    SPLIT(mozfun.map.get_key(extra, "results"), ',')[OFFSET(0)] AS result_type,
+    `mozfun.norm.result_type_to_product_name`(
+      SPLIT(mozfun.map.get_key(extra, "results"), ',')[OFFSET(0)]
+    ) AS product_result_type,
+    SPLIT(mozfun.map.get_key(extra, "results"), ',') AS results,
+    ARRAY(
+      SELECT
+        `mozfun.norm.result_type_to_product_name`(x)
+      FROM
+        UNNEST(SPLIT(mozfun.map.get_key(extra, "results"), ',')) AS x
+    ) AS product_results,
+    mozfun.map.get_key(extra, "selected_result") AS selected_result,
+    `mozfun.norm.result_type_to_product_name`(
+      mozfun.map.get_key(extra, "selected_result")
+    ) AS product_selected_result,
+    mozfun.map.get_key(extra, "engagement_type") AS engagement_type,
+    CAST(mozfun.map.get_key(extra, "n_chars") AS int) AS num_chars_typed,
+    CAST(mozfun.map.get_key(extra, "n_results") AS int) AS num_total_results,
+    metrics,
+    metrics.uuid.legacy_telemetry_client_id AS legacy_telemetry_client_id,
+    ping_info.experiments
+  FROM
+    `moz-fx-data-shared-prod.firefox_desktop_stable.events_v1`,
+    UNNEST(events)
+  WHERE
+    DATE(submission_timestamp) = @submission_date
+    AND category = 'urlbar'
+    AND name IN ('engagement', 'abandonment')
+),
+--remove events where the urlbar dropdown menu remains open (i.e., the urlbar session did not end)
+intermediate_states_removed AS (
+  SELECT *
+  FROM events_unnested
+  WHERE NOT (selected_result = 'tab_to_search' AND engagement_type in ('click', 'enter'))
+  AND NOT (selected_result = 'tip_dismissal_acknowledgement' AND engagement_type in ('click', 'enter'))
+  AND NOT (engagement_type in ('dismiss', 'inaccurate_location', 'not_interested', 'not_relevant', 'show_less_frequently'))
+)
```

Comment:
@dzeber This is based on our Slack DMs with Drew. What do you think about this implementation? Separately, we would create another table called 'urlbar_annoyance_signals'.

Reply:
I think it's fine to have this logic here in the query as it should only be used once.

I'm still wondering about having a separate table for non-terminal interactions, as opposed to an `is_terminal` column.
Pros:
- the number of search sessions is `count(*)` from the search sessions table
- can have separate schemas for the 2 tables if necessary

Cons:
- annoyances may be split across 2 tables (`help` is terminal)
- need to join tables to compute annoyance rate (num annoyance engagements / num search sessions)
- need to maintain 2 separate queries

Reply:
@dzeber ah - I forgot about the `is_terminal` option. I like that a bit better because we can define a Looker Measure to count rows if is_terminal is true. If we also rename this table to "urlbar_events" (or something not "urlbar_search_sessions"), then I'm good.

Reply:
Great - I pushed a commit with this change